dials_data/datasets.py

```diff
@@ -5,15 +5,19 @@
 import hashlib
 import os
 import textwrap
+from pathlib import Path
+from typing import Any
 
 import importlib_resources
-import py
 import yaml
 
 _hashinfo_formatversion = 1
 
+definition: dict[str, Any]
+fileinfo_dirty: set[str]
```

Comment:
```suggestion
definition: Dict[str, Any]
fileinfo_dirty: Set[str]
```

Reply:
No, we don't. Python 3.7 with `from __future__ import annotations` allows using standard collections, [PEP585](https://www.python.org/dev/peps/pep-0585/#implementation). This is why this repository now also enforces the `future` import in all Python files.

Reply:
Aha, I'd missed that. sorry.

Reply:
no worries. Time flies when you're supporting 2.7 :laughing:

Reply:
@ndevenish just pointed out that while this is valid type annotation it will break `typing.get_type_hints()` with Python <3.9.

Question: do we care?

Reply:
The consensus from https://github.com/cctbx/dxtbx/pull/463#issuecomment-1015308114 seems to be 'no', so let's go ahead with this.